front/app/modules/commercial/insights/hooks/useInsightsInputs.test.ts

```diff
@@ -0,0 +1,152 @@
+import useInsightsInputs, { QueryParameters } from './useInsightsInputs';
+import { Observable, Subscription } from 'rxjs';
+import { delay } from 'rxjs/operators';
+import { waitFor } from 'utils/testUtils/rtl';
+import { insightsInputsStream } from 'modules/commercial/insights/services/insightsInputs';
+
+const viewId = '1';
+
+const queryParameters: QueryParameters = {
+  category: '3',
+  pageSize: 10,
+  pageNumber: 12,
+  search: 'search',
+  sort: '-approval',
+};
+
+const mockInputs = {
+  data: {
+    currentPage: 1,
+    lastPage: 2,
+    list: [
+      {
+        id: '4e9ac1f1-6928-45e9-9ac9-313e86ad636f',
+        type: 'input',
+        relationships: {
+          source: {
+            data: {
+              id: '4e9ac1f1-6928-45e9-9ac9-313e86ad636f',
+              type: 'idea',
+            },
+          },
+          categories: {
+            data: [
+              {
+                id: '94a649b5-23fe-4d47-9165-9beceef2dcad',
+                type: 'category',
+              },
+              {
+                id: '94a649b5-23fe-4d47-9165-9becedfg45sd',
+                type: 'category',
+              },
+            ],
+          },
+          suggested_categories: {
+            data: [],
+          },
+        },
+      },
+      {
+        id: '54438f73-12f4-4b16-84f3-a55bd118de7e',
+        type: 'input',
+        relationships: {
+          source: {
+            data: {
+              id: '54438f73-12f4-4b16-84f3-a55bd118de7e',
+              type: 'idea',
+            },
+          },
+          categories: {
+            data: [],
+          },
+          suggested_categories: {
+            data: [],
+          },
+        },
+      },
+    ],
+  },
+};
+
+let mockObservable = new Observable((subscriber) => {
+  subscriber.next(mockInputs);
+}).pipe(delay(1));
+
+jest.mock('modules/commercial/insights/services/insightsInputs', () => {
+  return {
+    insightsInputsStream: jest.fn(() => {
+      return {
+        observable: mockObservable,
+      };
+    }),
+  };
+});
+
+describe('useInsightsInputs', () => {
+  it('should call useInsightsInputs with correct default arguments', async () => {
+    renderHook(() => useInsightsInputs(viewId));
+    expect(insightsInputsStream).toHaveBeenCalledWith(viewId, {
+      queryParameters: {
+        category: undefined,
+        'page[number]': 1,
+        'page[size]': 20,
+        search: undefined,
+        sort: 'approval',
+      },
+    });
+  });
+  it('should call useInsightsInputs with correct non-default arguments', async () => {
+    renderHook(() => useInsightsInputs(viewId, queryParameters));
+    expect(insightsInputsStream).toHaveBeenCalledWith(viewId, {
+      queryParameters: {
+        category: queryParameters.category,
+        'page[number]': queryParameters.pageNumber,
+        'page[size]': queryParameters.pageSize,
+        search: queryParameters.search,
+        sort: queryParameters.sort,
+      },
```

Comment:
Wouldn't 
```suggestion
      queryParameters
```
work ?

Reply:
No, because the keys for `page[number]` and `page[size]` are different. We are transforming them in the hook as well.